### PR TITLE
Changed getDescendants return value

### PR DESCRIPTION
**Pull request by Shahar-Y**
changed proto accordingly, added tests and some of the implementation of getFilesByFolder to match the rest of the code.

### CODE REVIEW

File: src/file/file.interface.ts [hunk 1]
```diff
@@ -35,20 +35,23 @@ export class IFile extends PrimitiveFile{
...
   createdAt?: Date;
   updatedAt?: Date;
+  size: number = 0;
 
   constructor(resFile: ResFile) {
     super(resFile);
-    if (resFile.createdAt) {
-      this.createdAt = new Date(resFile.createdAt);
+    // the '+' translates the string to int
+    if (+resFile.createdAt) {
```
Comment (avivbarel) [inline src/file/file.interface.ts]: cool stuff :+1:

File: src/file/file.service.ts [hunk 2]
```diff
@@ -220,30 +233,15 @@ export class FileService {
    * @param queryFile - the partial file containing the conditions.
    * @returns the pure query for extracting from the database.
    */
-  private static extractQuery(folderID: string | null, ownerID: string | null, queryFile?: Partial<IFile>): Partial<IFile> {
-    const parent = folderID ? new ObjectID(folderID) : null;
-    let query : Partial<IFile> = {};
-
-    // Create the query using the partial file,
+  private static extractQuery(queryFile?: Partial<IFile>): Partial<IFile> {
```
Comment (avivbarel) [inline src/file/file.service.ts]: Please consider refactoring to this function's documentation and naming.

File: src/file/file.service.ts [hunk 3]
```diff
@@ -253,11 +251,9 @@ export class FileService {
    * @param ownerID - owner of said folder.
    * @param query - the conditions.
    */
-  private static async getPopulatedChildren(folderID: string, ownerID: string, query: Partial<IFile>) : Promise<ResFile> {
-    const ancestor: ResFile = new ResFile(await this.getById(folderID));
+  private static async getPopulatedChildren(folderID: string, ownerID: string, query: Partial<IFile>) : Promise<ResFile[]> {
```
Comment (avivbarel) [inline src/file/file.service.ts]: This function can be removed and have it's name replacing the `_recursive` function name.

File: src/file/file.interface.ts (hunk 1 above)
Comment (Shahar-Y) [reply src/file/file.interface.ts]: can't take the credit.. @omrishtam

File: src/file/file.service.ts (hunk 2 above)
Comment (Shahar-Y) [reply src/file/file.service.ts]: right, I forgot to update it

File: src/file/file.service.ts (hunk 3 above)
Comment (Shahar-Y) [reply src/file/file.service.ts]: done

Comment (Shahar-Y) [reply src/file/file.service.ts]: done

**Pull request merged**